src/braket/ahs/analog_hamiltonian_simulation.py

```diff
@@ -0,0 +1,144 @@
+
+        Args:
+            register (AtomArrangement): The initial atom arrangement for the simulation.
+            hamiltonian (Hamiltonian): The hamiltonian to simulate.
+        """
+        self._register = register
+        self._hamiltonian = hamiltonian
+
+    @property
+    def register(self) -> AtomArrangement:
+        return self._register
+
+    @property
+    def hamiltonian(self) -> Hamiltonian:
+        return self._hamiltonian
+
+    def to_ir(self) -> ir.Program:
+        return ir.Program(
+            setup=ir.Setup(ahs_register=self._register_to_ir()),
+            hamiltonian=self._hamiltonian_to_ir(),
+        )
+
+    def _register_to_ir(self) -> ir.AtomArrangement:
+        return ir.AtomArrangement(
+            sites=[site.coordinate for site in self.register],
+            filling=[1 if site.site_type == SiteType.FILLED else 0 for site in self.register],
+        )
+
+    def _hamiltonian_to_ir(self) -> ir.Hamiltonian:
+        terms = defaultdict(list)
+        for term in self.hamiltonian.terms:
+            term_type, term_ir = _get_term_ir(term)
+            terms[term_type].append(term_ir)
+        return ir.Hamiltonian(
+            drivingFields=terms["driving_fields"], shiftingFields=terms["shifting_fields"]
+        )
+
+    def discretize(self, device) -> AnalogHamiltonianSimulation:
+        """Creates a new AnalogHamiltonianSimulation with all numerical values represented
+        as Decimal objects with fixed precision based on the capabilities of the device.
+
+        Args:
+            device (AwsDevice): The device for which to discretize the program.
+
+        Returns:
+            AnalogHamiltonianSimulation: A discretized version of this program.
+
+        Raises:
+            DiscretizeError: If unable to discretize the program.
+        """
+
+        required_action_schema = "braket.ir.ahs.program"
+        if (required_action_schema not in device.properties.action) or (
+            device.properties.action[required_action_schema].actionType != required_action_schema
+        ):
+            raise DiscretizationError(
+                f"AwsDevice {device} does not accept {required_action_schema} action schema."
+            )
+
+        properties = DiscretizationProperties(
+            device.properties.paradigm.lattice, device.properties.paradigm.rydberg
+        )
+        discretized_register = self.register.discretize(properties)
+        discretized_hamiltonian = self.hamiltonian.discretize(properties)
+        return AnalogHamiltonianSimulation(
+            register=discretized_register, hamiltonian=discretized_hamiltonian
+        )
+
+
+@singledispatch
+def _get_term_ir(
+    term: Hamiltonian,
+) -> Tuple[str, dict]:
+    raise TypeError(f"Unable to convert Hamiltonian term type {type(term)}.")
+
+
+@_get_term_ir.register
+def _(term: ShiftingField) -> Tuple[str, ir.ShiftingField]:
+    return "shifting_fields", ir.ShiftingField(
```

Comment:
Is the constant string "shifting_fields" defined somewhere that could be reused ? (Would help prevent errors arising due to typos, both for us and customers)